src/add-text-track-data.js

```diff
@@ -66,19 +66,38 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
   if (metadataArray) {
     metadataArray.forEach(function(metadata) {
       let time = metadata.cueTime + this.timestampOffset;
+      let videoDuration = Number.isFinite(this.mediaSource_.duration) ?
+        this.mediaSource_.duration :
+        Number.MAX_VALUE;
 
       metadata.frames.forEach(function(frame) {
         let cue = new Cue(
-            time,
-            time,
-            frame.value || frame.url || frame.data || '');
+          time,
+          videoDuration,
+          frame.value || frame.url || frame.data || '');
 
         cue.frame = frame;
         cue.value = frame;
         deprecateOldCue(cue);
+
         this.metadataTrack_.addCue(cue);
       }, this);
     }, sourceHandler);
+
+    /** Updating the metadeta cues so that
+     * the endTime of each cue is the startTime of the next cue
+     * the endTime of last cue is the duration of the video
+     */
+    if (sourceHandler.metadataTrack_ && sourceHandler.metadataTrack_.cues) {
+      for (let i = 0; i < sourceHandler.metadataTrack_.cues.length - 1; i++) {
+
+        if (sourceHandler.metadataTrack_.cues[i].endTime !==
+          sourceHandler.metadataTrack_.cues[i + 1].startTime) {
+          sourceHandler.metadataTrack_.cues[i].endTime =
+            sourceHandler.metadataTrack_.cues[i + 1].startTime;
+        }
+      }
+    }
```

Comment:
I think we will need to add code to `endOfStream` that will set the endpoint of the final cue in the `cues` array to have an `endTime` that is equal to the video's duration.